components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -311,6 +311,21 @@ void JsonRpcService::UpdateIsEip1559(const std::string& chain_id,
       if (!id || *id != chain_id)
         continue;
 
+      // Disable EIP-1559 on selected networks.
+      //
+      // This list needs to be updated until we have a generic EIP-1559 gas
+      // oracle. See: https://github.com/brave/brave-browser/issues/20469.
+      if (chain_id == "0x13881" ||  // Polygon Mumbai Testnet
+          chain_id == "0x89" ||     // Polygon Mainnet
+          chain_id == "0xa86a" ||   // Avalanche Mainnet
+          chain_id == "0xa869"      // Avalanche Fuji Testnet
+      ) {
+        changed =
+            custom_network.FindBoolKey("is_eip1559").value_or(false) != false;
```

Comment:
I think you can simplify to just 
`changed =  custom_network.FindBoolKey("is_eip1559").value_or(false);`

Reply:
Another thought, how about we just overwrite is_eip1559 to false for these networks in the very beginning (after `if (error != mojom::ProviderError::kSuccess)` check in line 297 with clear comments.
Then we won't need any special handling in later codes.
This seems better to me, like for example right now you're still calling `observer->OnIsEip1559Changed(chain_id, is_eip1559);` in line 342 with is_eip1559 instead of false.

Reply:
Also, we should create a new unit test case for this which covers both input eip_1559 = true and input eip_1559 = false, in the test case we should verify the observer is being called with expected value, and the pref is set to the expected value.

Reply:
Thanks, I incorporated the changes suggested by @yrliou and performed some tests locally as well.